### PR TITLE
Derivation adjustments

### PR DESCRIPTION
**Pull request by sebasguts**
@sebastianpos Please have a look if those commits solve your problems

### CODE REVIEW

File: CAP/gap/Derivations.gi [hunk 1]
```diff
@@ -348,28 +358,32 @@ InstallMethod( AddDerivationPair,
...
-    for current_implementation in Concatenation( implementations_with_extra_filters1, implementations_with_extra_filters2 ) do
-        
-        current_list := CAP_INTERNAL_FIND_APPEARANCE_OF_SYMBOL_IN_FUNCTION( current_implementation[ 1 ], operations_in_graph, loop_multiplier );
-        current_list := List( current_list, i -> [ ValueGlobal( i[ 1 ] ), i[ 2 ] ] );
-        ## TODO: Maybe move the next line to AddDerivationPairToCAP
-        current_list := Concatenation( current_list, CAP_INTERNAL_FIND_APPEARANCE_OF_SYMBOL_IN_FUNCTION_FOR_MONOIDAL_CATEGORIES( current_implementation[ 1 ], loop_multiplier ) );
-        collected_list := CAP_INTERNAL_MERGE_PRECONDITIONS_LIST( collected_list, current_list );
-        
-    od;
+    if preconditions_complete = false then
```
Comment (sebastianpos) [inline CAP/gap/Derivations.gi]: Your documentation above says that every other value
than `true` will be interpreted as non-complete, not only
`false.`

Comment (sebasguts) [reply CAP/gap/Derivations.gi]: Thank you, yes. I changed the logic again afterwards. I will fix up the documentation.

File: CAP/gap/Finalize.gi [hunk 2]
```diff
@@ -162,8 +165,14 @@ InstallMethod( IsFinalized,
             if ForAll( current_final_derivation.can_compute, j -> CurrentOperationWeight( weight_list, NameFunction( j[ 1 ] ) ) < infinity ) and
               ForAll( current_final_derivation.cannot_compute, j -> CurrentOperationWeight( weight_list, NameFunction( j ) ) = infinity ) then
                 
-                Add( current_installs, i );
-                
+                current_tester_func := current_final_derivation.category_filter;
+
+                if IsFilter( current_tester_func ) and Tester( current_tester_func )( category ) and current_tester_func( category ) then
```
Comment (sebastianpos) [inline CAP/gap/Finalize.gi]: Is this also a bugfix of the previous version?
It appears to be the case that final derivations where tried to be executed
regardless of the filters, is that correct?

Comment (sebasguts) [reply CAP/gap/Finalize.gi]: Indeed, that is the case. Do you think this can cause problems? It is indeed a silent bugfix ;)

**Comment by sebastianpos** [comment]
There is still the call of the function 
`CAP_INTERNAL_DERIVATION_SANITY_CHECK( graph, derivation );`
in `Derivations.gi`
that prevents me from using CAP-operations in derivations that
are not listed as precoditions.

**Comment by sebasguts** [comment]
Ah, thanks for pointing that out. I will disable that one.

**Comment by sebasguts** [comment]
Fixed

**Pull request merged**